tensorflow_graphics/datasets/features/camera_feature.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2020 The TensorFlow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Lint as: python3
+"""Camera feature."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+from tensorflow_datasets import features
+
+from tensorflow_graphics.datasets.features import pose_feature
+
+
+class Camera(features.FeaturesDict):
+  """`FeatureConnector` for camera calibration (extrinsic and intrinsic).
+
+  During `_generate_examples`, the feature connector accepts as input any of:
+
+    * `parameter_dict:` A dictionary containing the extrinsic and instrinsic  parameters of the camera as:
+      - 'rotation': 3x3 rotation matrix with float32 values.
+      - 'translation': 3x1 translation vector.
+      - 'f': focal length of the camera (either single float32 value or tuple of float32 as (f_x, f_y).
+      - 'optical_center': Optical center of the camera in pixel coordinates as tuple (c_x, c_y)
+      Optional parameters:
+      - 'skew': float32 denoting the skew of the camera axes.
+      - 'aspect_ratio': float32 denoting the aspect_ratio, if single fixed focal length is provided.
+
+
+  Output:
+    A dictionary containing:
+
+    * 'pose': A `tensorflow_graphics.datasets.features.Pose` FetureConnector the 3D pose of the camera.
+    * 'K': A `float32` tensor with shape `[3,3]` denoting the calibration matrix.
+
+  Example:
+    Default values for skew (s) and aspect_ratio(a) are 0 and 1, respectively.
+
+    Full calibration matrix:
+      K = [[ f_x,   s, c_x ],
+           [   0, f_y, c_y ],
+           [   0,   0,  1  ]]
+
+    With same focal length:
+      K = [[ f,  s, c_x ],
+           [ 0, af, c_y ],
+           [ 0,  0,  1  ]]
+  """
+
+  def __init__(self):
+    super(Camera, self).__init__({
+      'pose': pose_feature.Pose(),
+      'K': features.Tensor(shape=(3, 3), dtype=tf.float32),
+    })
+
+  def encode_example(self, parameter_dict):
+    """Convert the given parameters into a dict convertible to tf example."""
+    REQUIRED_KEYS = ['R', 't', 'f', 'optical_center']
+    assert list(parameter_dict.keys()) >= REQUIRED_KEYS, "Missing keys in provided dictionary!"
+
+    features_dict = {'pose': self._feature_dict['pose'].encode_example({
+      'R': parameter_dict['R'],
+      't': parameter_dict['t']
+    })}
+    aspect_ratio = 1
+    skew = 0
+    if 'aspect_ratio' in parameter_dict.keys():
```

Comment:
`if 'aspect_ratio' in parameter_dict:`

and similar for other key membership below